py-polars/polars/config.py

```diff
@@ -1450,3 +1452,62 @@ def set_expr_depth_warning(cls, limit: int) -> type[Config]:
 
         os.environ["POLARS_MAX_EXPR_DEPTH"] = str(limit)
         return cls
+
+    @classmethod
+    def set_engine_affinity(
+        cls, engine: Literal["cpu", "gpu", "streaming"] | None
+    ) -> type[Config]:
+        """
+        Set which engine to use by default.
+
+        Parameters
+        ----------
+        engine : Literal["cpu", "gpu", "streaming"]
+            The default engine to use on all .collect() calls
```

Comment:
We should soften this guarantee. We will try to run on this engine, but don't guarantee it. Both streaming and gpu have cpu fallbacks.